Libplanet/Store/Trie/KeyBytes.cs

```diff
@@ -0,0 +1,164 @@
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="KeyBytes"/> instance from the given byte array.
+        /// </summary>
+        /// <param name="bytes">An immutable byte array to wrap.</param>
+        public KeyBytes(in ImmutableArray<byte> bytes)
+        {
+            _byteArray = bytes;
+        }
+
+        /// <summary>
+        /// The length of the byte array.
+        /// </summary>
+        public int Length => _byteArray.IsDefault ? 0 : ByteArray.Length;
+
+        /// <summary>
+        /// The immutable array of bytes.
+        /// </summary>
+        public ImmutableArray<byte> ByteArray => _byteArray.IsDefault
+            ? ImmutableArray<byte>.Empty
+            : _byteArray;
+
+        /// <summary>
+        /// The hexadecimal string representation of the byte array.
+        /// </summary>
+        public string Hex => _byteArray.IsDefaultOrEmpty
+            ? string.Empty
+            : ByteUtil.Hex(_byteArray);
+
+        /// <summary>
+        /// Compares two <see cref="KeyBytes"/> values.
+        /// </summary>
+        /// <param name="left">An operand.</param>
+        /// <param name="right">Another operand.</param>
+        /// <returns><see langword="true"/> if two values equal; otherwise <see langword="false"/>.
+        /// </returns>
+        public static bool operator ==(KeyBytes left, KeyBytes right) => left.Equals(right);
+
+        /// <summary>
+        /// Compares two <see cref="KeyBytes"/> values.
+        /// </summary>
+        /// <param name="left">An operand.</param>
+        /// <param name="right">Another operand.</param>
+        /// <returns><see langword="false"/> if two values equal; otherwise <see langword="true"/>.
+        /// </returns>
+        public static bool operator !=(KeyBytes left, KeyBytes right) => !left.Equals(right);
+
+        /// <summary>
+        /// Parses the given hexadecimal string as bytes and returns a new <see cref="KeyBytes"/>
+        /// instance.
+        /// </summary>
+        /// <param name="hex">A hexadecimal string which encodes bytes.</param>
+        /// <returns>A new <see cref="KeyBytes"/> instance.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the given <paramref name="hex"/>
+        /// string is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the length of the given
+        /// <paramref name="hex"/> string is an odd number.</exception>
+        /// <exception cref="FormatException">Thrown when the given <paramref name="hex"/> string is
+        /// not a valid hexadecimal string.</exception>
+        public static KeyBytes FromHex(string hex) =>
+            new KeyBytes(ByteUtil.ParseHex(hex));
+
+        /// <summary>
+        /// Converts to a mutable byte array.
+        /// </summary>
+        /// <returns>A new copy of mutable byte array.</returns>
+        public byte[] ToByteArray() => ByteArray.IsDefault
```

Comment:
I think `ToByteArray()` together with `ByteArray` property above might be a bit too confusing. 😶

Reply:
It might be confusing, but is also consistent with existing byte-wrapping value types: `Address`, `TxId`, `HashDigest<T>`, `BlockHash`, etc.  I agree such naming is misleading (it might be better to rename it to `ToArray()`), but I don't want to expand the scope of this patch.  Could you make a separate issue?

Reply:
Ah, I see. The scope of the issue is larger than I thought. Fair enough.